corehq/apps/app_manager/fixtures/mobile_ucr.py

```diff
@@ -277,10 +278,67 @@ def __call__(self, restore_state):
         }
 
         if needed_versions.intersection({MOBILE_UCR_MIGRATING_TO_2, MOBILE_UCR_VERSION_2}):
-            fixtures.extend(self._v2_fixtures(restore_user, self._get_report_configs(apps).values()))
+            report_configs = self._get_report_configs(apps).values()
+            synced_fixtures, purged_fixture_ids = self._relevant_report_configs(restore_state, report_configs)
+            fixtures.extend(self._v2_fixtures(restore_user, synced_fixtures))
+            for report_uuid in purged_fixture_ids:
+                fixtures.extend(self._empty_v2_fixtures(report_uuid))
 
         return fixtures
 
+    def _relevant_report_configs(self, restore_state, report_configs):
+        """
+        Filter out any UCRs that are already synced. This can't exist in V1,
+        because in V1 we send all reports as one fixture.
+
+        Returns a list of full ReportConfigs to sync and a set of report ids to purge
```

Comment:
This feels a bit weird of a return value, but we need the full report config later and if an reports were deleted we only have the ID in the synclog